src/gui/guiFormSpecMenu.cpp

```diff
@@ -3098,22 +3098,36 @@ void GUIFormSpecMenu::regenerateGui(v2u32 screensize)
 			// different and forms should always use the
 			// maximum screen space available.
 			double prefer_imgsize = mydata.screensize.Y / 10 * gui_scaling;
-			double fitx_imgsize = mydata.screensize.X /
-				((12.0 / 8.0) * (0.5 + mydata.invsize.X));
-			double fity_imgsize = mydata.screensize.Y /
-				((15.0 / 11.0) * (0.85 + mydata.invsize.Y));
-			use_imgsize = MYMIN(prefer_imgsize,
-					MYMIN(fitx_imgsize, fity_imgsize));
+			if (mydata.real_coordinates) {
+				use_imgsize = std::min(prefer_imgsize,
+						(double)std::min(mydata.screensize.X / mydata.invsize.X,
+						mydata.screensize.Y / mydata.invsize.Y));
+			} else {
+				double fitx_imgsize = mydata.screensize.X /
+					((12.0 / 8.0) * (0.5 + mydata.invsize.X));
+				double fity_imgsize = mydata.screensize.Y /
+					((15.0 / 11.0) * (0.85 + mydata.invsize.Y));
+				use_imgsize = std::min(prefer_imgsize,
+						std::min(fitx_imgsize, fity_imgsize));
+			}
 #else
 			double prefer_imgsize = mydata.screensize.Y / 15 * gui_scaling;
-			double fitx_imgsize = mydata.screensize.X /
-				((5.0 / 4.0) * (0.5 + mydata.invsize.X));
-			double fity_imgsize = mydata.screensize.Y /
-				((15.0 / 13.0) * (0.85 * mydata.invsize.Y));
-			double screen_dpi = RenderingEngine::getDisplayDensity() * 96;
-			double min_imgsize = 0.3 * screen_dpi * gui_scaling;
-			use_imgsize = MYMAX(min_imgsize, MYMIN(prefer_imgsize,
-				MYMIN(fitx_imgsize, fity_imgsize)));
+			// In real_coordinates mode, the preferred size is the
+			// same as normal, but the form size will never exceed
+			// the screen size.
+			if (mydata.real_coordinates) {
+				use_imgsize = std::min(prefer_imgsize,
+						(double)std::min(mydata.screensize.X / mydata.invsize.X,
+						mydata.screensize.Y / mydata.invsize.Y));
+			} else {
+				double fitx_imgsize = mydata.screensize.X /
+					((5.0 / 4.0) * (0.5 + mydata.invsize.X));
+				double fity_imgsize = mydata.screensize.Y /
+					((15.0 / 13.0) * (0.85 * mydata.invsize.Y));
+				double screen_dpi = RenderingEngine::getDisplayDensity() * 96;
```

Comment:
This variable already exists on L3069.

Reply:
I wonder why GCC didn't give me an unused variable warning since it's no longer used...